argoverse/evaluation/detection_utils.py

```diff
@@ -0,0 +1,135 @@
+
+    if range_metric == "euclidean":
+        centers = np.array([dt.translation for dt in annos])
+        filtered_annos = np.array([])
+
+        if centers.shape[0] > 0:
+            dt_dists = np.linalg.norm(centers, axis=1)
+            filtered_annos = annos[dt_dists < max_dist]
+
+        return filtered_annos
+
+
+def get_ranks(dt_annos: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Get the rankings for the detection annotations.
+
+    Args:
+        dt_annos: Detection annotations.
+
+    Returns:
+        scores: The detection scores.
+        ranks: The ranking for the detections.
+
+    """
+    scores = np.array([dt_anno.score for dt_anno in dt_annos])
+    ranks = scores.argsort()[::-1]
+    return np.expand_dims(scores, 1)[ranks], ranks
+
+
+def interp(prec: np.ndarray, method="all") -> np.ndarray:
+    """Interpolate the precision over all recall levels.
+
+    Args:
+        prec: Precision at all recall levels.
+        method: Accumulation method.
+    
+    Returns:
+        Interpolated precision at all recall levels.
+    """
+    if method == "all":
+        return np.maximum.accumulate(prec[::-1])[::-1]
+
+
+def sim_fn(dt_annos: np.ndarray, gt_annos: np.ndarray, metric: SimFnType) -> np.ndarray:
+    if metric == SimFnType.CENTER:
+        dt_centers = np.array([dt.translation for dt in dt_annos])
+        gt_centers = np.array([gt.translation for gt in gt_annos])
+        sims = -cdist(dt_centers, gt_centers)
+    elif metric == SimFnType.IOU_2D:
+        raise NotImplemented("This similarity metric is not implemented!")
+    elif metric == SimFnType.IOU_3D:
+        raise NotImplemented("This similarity metric is not implemented!")
+    else:
+        raise NotImplemented("This similarity metric is not implemented!")
+    return sims
+
+
+def get_error_types(match_scores: np.ndarray, thresh: float, metric: SimFnType) -> np.ndarray:
+    # Euclidean distance represented as a "similarity" metric.
+    if metric == SimFnType.CENTER:
+        return match_scores > -thresh
+    else:
+        raise NotImplemented("This similarity metric is not implemented!")
+
+
+def dist_fn(dt_df: pd.DataFrame, gt_df: pd.DataFrame, metric: DistFnType) -> np.ndarray:
+    if metric == DistFnType.TRANSLATION:
+        dt_centers = np.vstack(dt_df["translation"].array)
+        gt_centers = np.vstack(gt_df["translation"].array)
+        trans_errors = np.linalg.norm(dt_centers - gt_centers, axis=1)
+        return trans_errors
+    elif metric == DistFnType.SCALE:
+        dt_dims = dt_df[["width", "length", "height"]]
+        gt_dims = gt_df[["width", "length", "height"]]
+        inter = np.minimum(dt_dims, gt_dims).prod(axis=1)
+        union = np.maximum(dt_dims, gt_dims).prod(axis=1)
+        scale_errors = 1 - (inter / union)
+        return scale_errors
+    elif metric == DistFnType.ORIENTATION:
+        dt_yaws = R.from_quat(np.vstack(dt_df["quaternion"].array)[:, [3, 0, 1, 2]]).as_euler("xyz")[:, 2]
```

Comment:
instead of repeating the [3,0,1,2] here, i would prefer to make the quat2yaw (or a better name) an explicit function in `argoverse.utils.transform`. I like how you are using scipy for all of this. you could use [this hacky function](https://github.com/johnwlambert/waymo_to_argoverse/blob/master/transform_utils.py#L29) as a unit test on it:

Reply:
I agree. Let's settle the `rotvec` portion and this simultaneously.